Dockerfile

```diff
@@ -1,59 +1,84 @@
 # ------------------------------------------------------------------------------
-# dependancy stage
+# base
 # ------------------------------------------------------------------------------
-FROM mkenney/npm:latest AS dependencies
-ENV INSTALL_PATH /deps
-RUN mkdir -p $INSTALL_PATH
-WORKDIR $INSTALL_PATH
-COPY package.json ./package.json
-COPY package-lock.json ./package-lock.json
-RUN npm set progress=false && npm config set depth 0
-RUN npm install --save govuk-frontend
-
-# ------------------------------------------------------------------------------
-# release stage
-# ------------------------------------------------------------------------------
-
-FROM ruby:2.6.3 as release
+FROM ruby:2.6.3 AS base
 MAINTAINER dxw <rails@dxw.com>
+
 RUN apt-get update && apt-get install -qq -y \
   build-essential \
   libpq-dev \
   --fix-missing --no-install-recommends
-RUN curl -sL https://deb.nodesource.com/setup_10.x | bash - \
-        && apt-get install -y nodejs
 
-ENV INSTALL_PATH /srv/beis-roda
-RUN mkdir -p $INSTALL_PATH
+ENV APP_HOME /app
+ENV DEPS_HOME /deps
 
-# This must be ordered before rake assets:precompile
-COPY --from=dependencies ./deps/node_modules $INSTALL_PATH/node_modules
-COPY --from=dependencies ./deps/node_modules/govuk-frontend/govuk/assets $INSTALL_PATH/app/assets
+ARG RAILS_ENV
+ENV RAILS_ENV ${RAILS_ENV:-production}
+ENV NODE_ENV ${RAILS_ENV:-production}
 
-WORKDIR $INSTALL_PATH
+# ------------------------------------------------------------------------------
+# dependancies
+# ------------------------------------------------------------------------------
+FROM base AS dependencies
 
-# set rails environment
-ARG RAILS_ENV
-ENV RAILS_ENV=${RAILS_ENV:-production}
-ENV RACK_ENV=${RAILS_ENV:-production}
+# Set up install environment
+RUN mkdir -p ${DEPS_HOME}
+WORKDIR ${DEPS_HOME}
+# End
 
-COPY Gemfile $INSTALL_PATH/Gemfile
-COPY Gemfile.lock $INSTALL_PATH/Gemfile.lock
+RUN curl -sL https://deb.nodesource.com/setup_10.x | bash - \
+        && apt-get install -y nodejs
+
+# Install Ruby dependencies
+COPY Gemfile ${DEPS_HOME}/Gemfile
+COPY Gemfile.lock ${DEPS_HOME}/Gemfile.lock
 
 RUN gem update --system 3.0.3
 RUN gem update rake 13.0.1
 RUN gem install bundler
 
-# bundle ruby gems based on the current environment, default to production
-RUN echo $RAILS_ENV
-RUN \
-  if [ "$RAILS_ENV" = "production" ]; then \
-    bundle install --jobs 4 --without development test --retry 10; \
+RUN bundle config set frozen 'true'
```

Comment:
What does this do?

Reply:
This is section is lifted out of Claims Dockerfile, my goal was to get rid of the `--without` as it's deprecated, however that involves the new style bundle config so I kept this, my understanding is that `frozen` tells bunder to not change the Gemfile and that would mean if the Gemfile and .lock don't match we get an error - the other case is the Gemfile getting re-written based off the .lock without us knowing – I will admit my knowledge is weak here!

Looking at the docs, it is a little confusing to me:

https://bundler.io/v2.0/man/bundle-config.1.html

`frozen` seems to imply `deployment` but perhaps not `--deployment` - I'd like to understand this more, this explains things a little better:

https://bundler.io/guides/deploying.html#deploying-your-application

This makes me think that we should drop the `frozen` and instead use `--deployment` on the bundle install (which includes `frozen`) - but then I think we may have to use `bundle exec` everywhere? (see your later comment)